cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -160,19 +166,22 @@ public void enqueue(String accountId, String stream, Iterator<? extends StreamEv
    * @throws IOException if fails to get stream information
    * @throws IllegalArgumentException If the stream doesn't exists
    */
-  public void asyncEnqueue(String accountId, final String stream,
-                           Map<String, String> headers, ByteBuffer body, Executor executor) throws IOException {
+  public ListenableFuture<Long> asyncEnqueue(String accountId, final String stream,
```

Comment:
How is it being used in async case? The server doesn't block for the result to return.